### PR TITLE
GN-4450: Internationalization of variable plugins using document lang

### PR DESCRIPTION
**Pull request by dkozickis**
### Overview

Internationalization of variable plugins using document language. 
Builds on top of work done by @elpoelma 🙏 

#### Connected issues and PRs:

* https://binnenland.atlassian.net/browse/GN-4450

### Setup

1. Checkout
2. `npm run start`

### How to test/reproduce

* Check `number` variable
* Check `address` variable
* Check serialization (e.g. in the CustomHTML debug helper)

CleanShot 2023-09-13 at 13 17 20@2x

CleanShot 2023-09-13 at 13 17 40@2x

### Checks PR readiness
- [x] UI: works on smaller screen sizes
- [x] UI: feedback for any loading/error states
- [x] Check if dummy app is correctly updated
- [x] Check cancel/go-back flows
- [x] changelog
- [x] npm lint
- [x] no new deprecations

### CODE REVIEW

**Review by elpoelma — CHANGES_REQUESTED** [review]
The translation of the placeholders looks good.

- I'm not sure if we should translate the popup interface to edit the number variable based on the document language (as that interface is not really part of the document).

- I'm thinking we might also need to translate the default variable labels based on document language.

**Comment by elpoelma** [comment]
> > * I'm thinking we might also need to translate the default variable labels based on document language.
> 
> Done 🙏
> 
> > * I'm not sure if we should translate the popup interface to edit the number variable based on the document language (as that interface is not really part of the document).
> 
> Rolled that part back
> 
> For the date plugin translation - I have resorted to using `outlineText` for date spec definition, so it can be shown as text in TOC, if anyone has a config like that 😅 Because we have agreed to remove `options.insertDate` translations for date plugin, I have resorted to using default translations in `parseDOM` and `leafText` of date plugin. I guess we need `leafTextSerialize` and `parseDOMSerialize` alternatives to be able to provide correct translations there 😅

I agree that a `leafTextSerialize` could be useful, but why would a `parseDOMSerialize` be needed?

**Comment by x-m-el** [comment]
@dkozickis see #298, which added posibility for address label.
As far as I can see it should just work, but test the address variable label translation specifically when merging with master, just to be sure my changes didn't mess up something :)

**Comment by elpoelma** [comment]
> > I agree that a `leafTextSerialize` could be useful, but why would a `parseDOMSerialize` be needed?
> 
> E.g. for this bit - https://github.com/lblod/ember-rdfa-editor-lblod-plugins/pull/297/files#diff-d28e2d121d03a4ca9f4e4ef7e53669c829221b162b507bb7ced8b4f683530d6cR163

Aha I see, but I think we could also decide which placeholder to show in the component, where we have access to internationalization (instead of in `parseDOM`).

**Comment by elpoelma** [comment]
> I agree with this.
> 
>     * Pop-up: purely UI for the user, can follow their language
> 
>     * variable label: these might be added by different people at different stages of the document. To have the biggest chance of them being consistent language, they should follow the document language.
> 
> 
> @elpoelma the "title" of a template comment follows the browser-language, but thinking about it, it might make sense to follow document language? It is "UI" as it tells what the blue block is, but it looks like part of the text (you might read it while reading the document). So to avoid having something suddenly English inbetween a Dutch text, I feel like that should also follow document language?
> 
> Maybe a general way to approach this (although there might still be edge cases): Text that is always on the document should be follow document-language (labels, template comment title, placeholder...) Text outside of document (UI) or pop-ups with explanation/errors/etc (also UI) should follow browser language.
> 
> Where in principle someone reading a document and not doing edits should only come across the document language and never the UI (browser) language.

Agree with this: text which is really inside the document should IMO always be translated based on the document language. So following this approach, the title of a template comment should indeed follow the document language (I think it's also included in the exported html).

**Review by elpoelma — CHANGES_REQUESTED** [review]
Seems to work pretty well overall.

On the variable default labels:
- I think we can remove the default value for the label attribute in the variable node-specs
- The translated default value is only calculated once (on insertion), I'm thinking we might need to translate this in the `serialize` function?

**Comment by elpoelma** [comment]
> When choosing a label yourself (not using the default), it will also be chosen at insertion and never change afterwards. So it being calculated at insertion once feels consistent.
> 
> Template comment title is indeed shown on export, so even more reason it should also follow document language :)

Yeah that's true, then I agree we should go for consistency.

File: translations/en-US.yaml [hunk 1]
```diff
@@ -283,6 +293,7 @@ template-comments-plugin:
     remove: Remove template comment or example
     move-up: Move comment up
     move-down: Move comment down
+  heading: explanation of provision
```
[the remark this answers is not in the document]
Comment (x-m-el) [reply translations/en-US.yaml:296]: using "template comment" is okay for English version. The Dutch version is that name to be "legally" correct, but really it is just a comment in everything but name :p

[the remark this answers is not in the document]
Comment (x-m-el) [reply translations/en-US.yaml:296]: Although for the document itself just "Comment or example" might be better. Referring to "template" when you are reading a document is a bit weird.

**Review by abeforgit — CHANGES_REQUESTED** [review]
this seems to break insertion of dates
image

**Comment by elpoelma** [comment]
> this seems to break insertion of dates image

IMO, we should keep the labels optional (with a default value of `null`), especially to take existing document with variables without labels into account.

**Comment by elpoelma** [comment]
> > IMO, we should keep the labels optional (with a default value of `null`), especially to take existing document with variables without labels into account.
> 
> Can we do that in a separate PR please? I keep updating this one, and then updating it again 😓

In understand, but the label not having a `null` default value was what broke the insertion of dates. This PR is breaking for existing documents with variables that don't have any labels (e.g. some date variables). So the parsing will break.

**Review by elpoelma — APPROVED** [review]
Works very nicely. Nice job! :smile: 
I think this PR is technically breaking as some of the parts of `DateOptions` are removed, so maybe we should add a `major` changeset for that part.

**Comment by elpoelma** [comment]
> > Works very nicely. Nice job! 😄 I think this PR is technically breaking as some of the parts of `DateOptions` are removed, so maybe we should add a `major` changeset for that part.
> 
> Updated changeset 🙏

super! :)

**Pull request merged**